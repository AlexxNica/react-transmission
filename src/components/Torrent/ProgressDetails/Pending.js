import React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';

import { size as formatSize } from 'util/formatters';

function ProgressPending({ torrent, intl }) {
  const { sizeWhenDone, leftUntilDone, percentDone } = torrent;

  return (
    <FormattedMessage
      id='torrent.progress.pending'
      defaultMessage={`{remaining} of {total} ({completed})`}
      values={{
        remaining: formatSize(sizeWhenDone - leftUntilDone),
        total: formatSize(sizeWhenDone),
        completed: intl.formatNumber(percentDone, {style: 'percent'}),
      }}
    />
  );
}

export default injectIntl(ProgressPending);
