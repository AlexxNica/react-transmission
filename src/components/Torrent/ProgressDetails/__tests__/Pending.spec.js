import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import Pending from '../Pending';

test('Pending', () => {
  const component = createComponentWithIntl(
    <Pending torrent={{sizeWhenDone: 1E5, leftUntilDone: 1E4, percentDone: 0.9}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
