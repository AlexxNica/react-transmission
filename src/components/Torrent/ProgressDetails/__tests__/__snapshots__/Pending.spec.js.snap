exports[`test Pending 1`] = `
<span>
  90.0 kB of 100.0 kB (90%)
</span>
`;
